.github/workflows/main.yml

```diff
@@ -12,28 +20,39 @@ jobs:
   buildtestandanalyze:
     uses: ./.github/workflows/build-test-analyze.yml
     secrets: inherit
-
-  uploadartifacts:
-    name: Archive production artifacts
+  googleplayalpha:
+    name: Upload App to Google Play - Alpha
     runs-on: ubuntu-latest
     needs: buildtestandanalyze
+    environment: alpha
     steps:
       - name: Checkout code
-        uses: actions/checkout@v4.1.0
-      - name: Upload artifacts
-        uses: actions/upload-artifact@v2
-        with:
-          name: app
-          path: app/build/outputs/apk/release/app-release-unsigned.apk
-
+        uses: actions/checkout@v4
+      - name: Bundle Install
+        run: |
+          gem install bundler
+          bundle install
```

Comment:
minor: so far i think all steps had only one single command that was executed. we could think about splitting this line into two steps: instann bundler and bundle install

Reply:
I think its fine if we keep corresponding steps in a single run execution, we do a similar thing in our reusable workflow and try to group them by "step", e.g., installing dependencies.